data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -290,8 +295,6 @@ private static Schema fileProjection(Schema tableSchema, Schema requestedSchema,
       requiredIds.addAll(eqDelete.equalityFieldIds());
     }
 
-    requiredIds.add(MetadataColumns.IS_DELETED.fieldId());
```

Comment:
Could you comment on why this was removed?

Reply:
We shouldn't add the metadata column if user doesn't ask a project or filter on it, right?  
It was added by #2538. Hi @chenjunjiedada, any idea why we need this line in #2538?

Reply:
We already project an extra position column, right? It depends on whether the projected column is useful. The projected  `_delete` column can be used for converting delete as I wrote if I recall. The CDC changes may also need it for joining, right?

Reply:
We project the pos column only if there are pos deletes as the following code shows, which makes sense, since we need it for filtering pos deletes. 
```
    if (!posDeletes.isEmpty()) {
      requiredIds.add(MetadataColumns.ROW_POSITION.fieldId());
    }
```
Here is my thought on Is_deleted column, it presents only if the front end(e.g. spark read) asked for it. For example, in case of CDC, we put it filter like this to read deleted rows. Here is the code from my CDC draft PR #4539.
```
    Dataset<Row> scanDF = spark().read().format("iceberg")
        .option(SparkReadOptions.FILE_SCAN_TASK_SET_ID, groupID)
        .load(table.name())
        .filter(functions.column(MetadataColumns.IS_DELETED.name()).equalTo(true));
```

Reply:
Sounds good to me!